src/Fixer/Semicolon/SemicolonOnNewLineForChainedCallFixer.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Semicolon;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Egidijus Girčys <e.gircys@gmail.com>
+ */
+final class SemicolonOnNewLineForChainedCallFixer extends AbstractFixer implements WhitespacesAwareFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(';');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Semicolon must be on the new line for chained calls.',
+            [
+                new CodeSample('
+                    <?php
+                        $this->method1()
+                            ->method2()
+                            ->method(3);
+                    ?>
+'),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must run after the NoSinglelineWhitespaceBeforeSemicolonsFixer
+        return 10;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = 1, $count = count($tokens); $index < $count; ++$index) {
+            // continue if token is not a semicolon
+            if (!$tokens[$index]->equals(';')) {
+                continue;
+            }
+
+            // get the indent of the chained call, null in case it's not a chained call
+            $indent = $this->getIndentation($index - 1, $tokens);
+
+            if (null === $indent) {
+                continue;
+            }
+
+            // unset semicolon
+            $tokens->clearAt($index);
+
+            // find the line ending token index after the semicolon
+            $index = $this->getNewLineIndex($index, $tokens);
+
+            // line ending string of the last method call
+            $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+            // appended new line to the last method call
+            $newline = new Token([T_WHITESPACE, $lineEnding.$indent]);
+
+            // insert the new line with indented semicolon
+            $tokens->insertAt($index, [$newline, new Token(';')]);
```

Comment:
because of that, you do not iterate over whole collection.
https://3v4l.org/TipvA

imagine you added 100x whitespaces in different places, that means last 100 of tokens were not checked, bringing possibility that some occurrence were not fixed.

Taking into consideration that you are iterate forward in `applyFix` and then backwards in `getIndentation`, you shall probably iterate backward in `applyFix` as well. That way you would fix mentioned issue and don't visit same tokens twice.